simln-lib/src/lib.rs

```diff
@@ -476,9 +476,7 @@ pub struct Simulation {
     /// Config for the simulation itself.
     cfg: SimulationCfg,
     /// The lightning node that is being simulated.
-    nodes: HashMap<PublicKey, Arc<Mutex<dyn LightningNode>>>,
-    /// The activity that are to be executed on the node.
-    activity: Vec<ActivityDefinition>,
+    pub nodes: HashMap<PublicKey, Arc<Mutex<dyn LightningNode>>>,
```

Comment:
Don't love that we have to make this `pub`

Now that activity validation has been DRY-ed up, I think that we can have an in-between approach where the `create_simulation...` functions return them. Rather than exposing the client info to `main` just so we can validate the activites.

[Patch here](https://github.com/user-attachments/files/19801093/activities_on_create.patch) for suggestion, I think it hits a good middle ground.